src/databricks/labs/ucx/azure/credentials.py

```diff
@@ -87,6 +87,7 @@ def list(self, include_names: set[str] | None = None) -> set[str]:
     def create_with_client_secret(self, spn: ServicePrincipalMigrationInfo) -> StorageCredentialInfo:
         # prepare the storage credential properties
         name = spn.permission_mapping.principal
+        assert spn.permission_mapping.directory_id is not None
```

Comment:
this will crash the CLI command, right?

Reply:
Yes, technically it will if `create_with_client_secret` is called with a ServicePrincipalMigrationInfo that contains managed identity. But `create_with_client_secret` is only supposed to be called with ServicePrincipalMigrationInfo that contains a service principal.

Reply:
Next time - add code comment upfront 🙃